frontend/src/component/pages/shared/CreateShortLinkSection.tsx

```diff
@@ -7,77 +7,188 @@ import { ShortLinkUsage } from './ShortLinkUsage';
 import { Section } from '../../ui/Section';
 import { Url } from '../../../entity/Url';
 import { UIFactory } from '../../UIFactory';
+import { validateLongLinkFormat } from '../../../validators/LongLink.validator';
+import { validateCustomAliasFormat } from '../../../validators/CustomAlias.validator';
+import { raiseCreateShortLinkError } from '../../../state/actions';
+import { IAppState } from '../../../state/reducers';
+import { Store } from 'redux';
+import { UrlService } from '../../../service/Url.service';
+import { QrCodeService } from '../../../service/QrCode.service';
 
-interface Props {
+interface IProps {
+  store: Store<IAppState>;
   uiFactory: UIFactory;
-  longLinkText?: string;
+  urlService: UrlService;
+  qrCodeService: QrCodeService;
+  onShortLinkCreated?: (shortLink: string) => void;
+  onRequestSignIn?: () => void;
+}
+
+interface IState {
+  longLink: string;
   alias?: string;
-  shortLink?: string;
-  inputErr?: string;
-  createdUrl?: Url;
-  qrCodeUrl?: string;
+  inputError: string | null;
   isShortLinkPublic?: boolean;
-  onLongLinkTextFieldBlur?: () => void;
-  onLongLinkTextFieldChange?: (newLongLink: string) => void;
-  onShortLinkTextFieldBlur?: () => void;
-  onShortLinkTextFieldChange?: (newAlias: string) => void;
-  onPublicToggleClick?: (enabled: boolean) => void;
-  onCreateShortLinkButtonClick?: () => void;
+  shouldShowUsage: boolean;
```

Comment:
Just curious if we can make this a function instead. One less state to maintain and it is dependent on the existence of the below 3 states.
```
private shouldShowUsage() {
    return (this.state.createdLongLink && this.state.createdShortLink && this.state.qrCodeURL);
}
```